### ISSUE TITLE
Cannot convert from Typed Mat_<T> to &mut dyn ToInputOutputArray

### ISSUE THREAD

**Issue by ryanseipp** [twistedfall/opencv-rust#370]
1. Operating system:
Arch Linux kernel 5.19.7-arch1-1
2. The way you installed OpenCV: package, official binary distribution, manual compilation, etc.
Pacman package
3. OpenCV version
4.6.0-4
4. rustc version (`rustc --version`)
rustc 1.63.0 (4b91a6ea7 2022-08-08)

This is more of a question/discussion about the exposed API rather than a bug per-se.
As I understand it, the following is not possible:
```rs
fn main() -> Result<(), dyn std::error::Error> {
    let mut image: Mat1b = Mat::zeros(3, 3, CV_8UC1)?
        .to_mat()?
        .try_into_typed()?;

    line(&mut image, Point::new(0.0, 1.0), Point::new(1.0, 1.0), 1.0, LINE_4, 0)?;
}
```
as the following diagnostic is returned
```
the trait bound `Mat_<u8>: _InputOutputArrayTrait` is not satisfied
the trait `_InputOutputArrayTrait` is implemented for `_InputOutputArray`
required because of the requirements on the impl of `ToInputOutputArray` for `Mat_<u8>`
required for the cast to the object type `dyn ToInputOutputArray`
```
However, if I change the declaration of `image` to the following
```rs
let mut image = Mat::zeros(3, 3, CV_8UC1)?.to_mat()?;
```
everything compiles successfully. Unfortunately `Mat_<T>.as_untyped()` doesn't help us here, as the function takes `&self` and has no `&mut self` variant, and I don't see any other way of converting a `mut Mat_<T>` to a `mut Mat` to get around this (unless of course I've missed something).

This seems like an opportunity to improve the API, making the functionality of typed `Mat_<T>` and untyped `Mat` more compatible. I'm unaware if there's technical reasons preventing this from happening, or if it just takes the time to work through.

**Comment by twistedfall** [comment]
Hi, thank you for the detailed report!

> any other way of converting a mut Mat_<T> to a mut Mat to get around this (unless of course I've missed something).

There is also [into_untyped](https://docs.rs/opencv/0.67.0/opencv/core/struct.Mat_.html#method.into_untyped) which should help here. Otherwise what you suggest makes perfect sense, I'll check it out why it's not implemented.

**Comment by twistedfall** [comment]
There was actually no reason for that, it will be fixed in the next release

**Comment by twistedfall** [comment]
Fixed in v0.68.0
